ModManagerUI/ModsBox.cs

```diff
@@ -0,0 +1,228 @@
+using Modio.Filters;
+using Modio.Models;
+using ModManager;
+using ModManager.AddonSystem;
+using ModManager.ModIoSystem;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Timberborn.AssetSystem;
+using Timberborn.CoreUI;
+using UnityEngine;
+using UnityEngine.UIElements;
+using Image = UnityEngine.UIElements.Image;
+using TextField = UnityEngine.UIElements.TextField;
+
+namespace Timberborn.ModsSystemUI
+{
+    public class ModsBox : IPanelController
+    {
+
+
+        private VisualElement LoadVisualElement(string elementName) => LoadVisualElement(LoadVisualTreeAsset(elementName));
+        private VisualTreeAsset LoadVisualTreeAsset(string elementName) => _resourceAssetLoader.Load<VisualTreeAsset>($"{elementName}");
+
+        private VisualElement LoadVisualElement(VisualTreeAsset visualTreeAsset)
+        {
+            VisualElement visualElement = visualTreeAsset.CloneTree().ElementAt(0);
+            _visualElementInitializer.InitializeVisualElement(visualElement);
+            return visualElement;
+        }
+
+
+        public static Action OpenOptionsDelegate;
+        private readonly VisualElementLoader _visualElementLoader;
+        private readonly PanelStack _panelStack;
+        private readonly IAddonService _addonService;
+        private readonly VisualElementInitializer _visualElementInitializer;
+        private readonly IResourceAssetLoader _resourceAssetLoader;
+        private VisualElement _mods;
+        private Filter _filter = ModFilter.Downloads.Desc();
+        private Label _loading;
+        private Label _error;
+        private TextField _search;
+        private RadioButtonGroup _tags;
+        private List<string> _tagOptions;
+        private ExtractorService _extractor;
+
+        private AssetBundle _bundle;
+
+        public ModsBox(VisualElementLoader visualElementLoader,
+                       PanelStack panelStack,
+                       IAddonService addonService,
+                       VisualElementInitializer visualElementInitializer,
+                       IResourceAssetLoader resourceAssetLoader,
+                       ExtractorService extractor)
+        {
+            _visualElementLoader = visualElementLoader;
+            _panelStack = panelStack;
+            _addonService = addonService;
+            OpenOptionsDelegate = OpenOptionsPanel;
+            _visualElementInitializer = visualElementInitializer;
+            _resourceAssetLoader = resourceAssetLoader;
+            _extractor = extractor;
+
+            _bundle = AssetBundle.LoadFromFile(@"D:\Ohjelmat\Steam\steamapps\common\Timberborn\BepInEx\plugins\ModManager\assets\what.bundle");
```

Comment:
Don't forget to chane to a dynamic path with Path.Combine

Reply:
👍